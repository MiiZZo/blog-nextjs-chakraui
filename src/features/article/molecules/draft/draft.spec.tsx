import React from "react";
import { cleanup } from "@testing-library/react";
import { myRender } from "@lib/testing/render";
import Draft from "./draft";

afterEach(cleanup);

describe("draft component", () => {
  const handleChangeDraftData = jest.fn();
  const handleSubmit = jest.fn();
  const handleToggleShowPreview = jest.fn();

  it("should match a shapshot", () => {
    const { asFragment } = myRender(
      <Draft
        draftData={{
          title: "title",
          body: "body",
          description: "description"
        }}
        handleChangeDraftData={handleChangeDraftData}
        handleSubmit={handleSubmit}
        handleToggleShowPreview={handleToggleShowPreview}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
