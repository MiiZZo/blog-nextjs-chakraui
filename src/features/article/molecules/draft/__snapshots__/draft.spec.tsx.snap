// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`draft component should match a shapshot 1`] = `
<DocumentFragment>
  <form
    class="css-17qrq1d"
  >
    <div
      class="css-8fkoyy"
      role="group"
    >
      <label
        class="css-176s5fq"
      >
        Title
      </label>
      <div
        class="css-kjafn5"
      >
        <input
          aria-invalid="false"
          class="css-dgww3q"
          data-testid="title-input"
          name="title"
          placeholder="Title"
          value="title"
        />
        <div
          class="css-1f4xxbf"
        >
          <div
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuenow="5"
            class="css-1bfqfh5"
            data-testid="circular-indicator"
            role="progressbar"
          >
            <svg
              class="css-24ggqe"
              viewBox="55.55555555555556 55.55555555555556 111.11111111111111 111.11111111111111"
            >
              <circle
                class="css-9y0bww"
                cx="111.11111111111111"
                cy="111.11111111111111"
                data-progress-track="true"
                r="50"
                stroke-dasharray="314.159"
                stroke-dashoffset="0"
                stroke-width="11.111111111111112"
              />
              <circle
                class="css-ich4kh"
                cx="111.11111111111111"
                cy="111.11111111111111"
                data-progress-indicator="true"
                r="50"
                stroke-dasharray="314.159"
                stroke-dashoffset="298.45130209103036"
                stroke-width="11.111111111111112"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-8fkoyy"
      role="group"
    >
      <label
        class="css-176s5fq"
      >
        Description
      </label>
      <div
        class="css-kjafn5"
      >
        <input
          aria-invalid="false"
          class="css-dgww3q"
          data-testid="description-input"
          name="description"
          placeholder="Description"
          value="description"
        />
        <div
          class="css-1f4xxbf"
        >
          <div
            aria-valuemax="100"
            aria-valuemin="0"
            aria-valuenow="11"
            class="css-1bfqfh5"
            data-testid="circular-indicator"
            role="progressbar"
          >
            <svg
              class="css-24ggqe"
              viewBox="55.55555555555556 55.55555555555556 111.11111111111111 111.11111111111111"
            >
              <circle
                class="css-9y0bww"
                cx="111.11111111111111"
                cy="111.11111111111111"
                data-progress-track="true"
                r="50"
                stroke-dasharray="314.159"
                stroke-dashoffset="0"
                stroke-width="11.111111111111112"
              />
              <circle
                class="css-ich4kh"
                cx="111.11111111111111"
                cy="111.11111111111111"
                data-progress-indicator="true"
                r="50"
                stroke-dasharray="314.159"
                stroke-dashoffset="279.6017461694916"
                stroke-width="11.111111111111112"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
    <div
      class="css-8fkoyy"
      role="group"
    >
      <label
        class="css-176s5fq"
      >
        Body (markdown)
      </label>
      <textarea
        aria-invalid="false"
        class="css-1tlg8wm"
        data-testid="body-input"
        name="body"
        placeholder="Body (markdown)"
      >
        body
      </textarea>
    </div>
    <div
      class="css-8fkoyy"
      role="group"
    >
      <button
        class="css-jifgjd"
        type="submit"
      >
        Publish article
      </button>
    </div>
    <button
      class="css-c4fa1y"
      data-testid="show"
      type="button"
    >
      Show preview
    </button>
  </form>
</DocumentFragment>
`;
